epochStart/shardchain/triggerRegistry.go

```diff
@@ -55,6 +52,26 @@ func (t *trigger) LoadState(key []byte) error {
 	return nil
 }
 
+// UnmarshalTrigger unmarshalls the trigger
+func (t *trigger) UnmarshalTrigger(data []byte) (*TriggerRegistry, error) {
+	state := &TriggerRegistry{
+		EpochStartShardHeader: &block.Header{},
```

Comment:
Maybe first we can try with &block.HeaderV2{} here and in line 66 with &block.Header{}, exactly as we did in CreateShardHeader, as after the point of the activation epoch, it should work with only one unmarshal

Reply:
done